.devcontainer/Dockerfile

```diff
@@ -1,22 +1,30 @@
 FROM mcr.microsoft.com/dotnet/sdk:3.1-focal
 
-# Install Mono and Nuget CLI
-RUN apt-get -y update && \
-    apt-get -y install dirmngr gnupg apt-transport-https ca-certificates && \
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
-    sh -c 'echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" > /etc/apt/sources.list.d/mono-official-stable.list' && \
-    apt-get -y update && \
-    apt-get -y install --no-install-recommends mono-complete nuget && \
-    apt-get clean && rm -rf /var/lib/apt/lists/
-
-RUN apt update \
-    && apt-get install -y cmake \
+# Install Mono, NuGet, CMake, Ninja, Clang-11, build-essential
+RUN apt-get -y update \
+    && apt-get -y install dirmngr gnupg apt-transport-https ca-certificates \
+    && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
+    && sh -c 'echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" > /etc/apt/sources.list.d/mono-official-stable.list' \
+    && apt-get -y update \
+    && apt-get -y install --no-install-recommends mono-complete nuget \
+    && apt-get -y install jq \
+                          cmake \
                           ninja-build \
                           clang-11 \
                           clang-tidy-11 \
                           build-essential \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/
+    && apt-get clean && rm -rf /var/lib/apt/lists/
+
+# Install latest CMake (required since apt-get uses 3.16 and repo requires 3.20)
+RUN json=$(curl -s "https://api.github.com/repos/Kitware/CMake/releases") \
```

Comment:
This somewhat horrifies me, but I confess that I don't have any suggestions on how to improve...!

Reply:
Yeah, it's a odd thing to include, but I copied it from the scripts that configure the Azure DevOps/GitHub Actions pipeline VMs (see https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/installers/cmake.sh). The good news is that when apt gets the option to install CMake 3.20 (either by default or as an explicit opt-in) then we can remove this.

Reply:
Makes perfect sense, my slight horror notwithstanding!